### ISSUE TITLE
Procedure Argument Type: Variable length

### ISSUE THREAD

**Issue by ddmills** [hridesh/panini#12]
We have not investigated whether any of these capsule procedure declarations work:

```java
public void foo(Object[] baz) {...}
public void foo(int[] baz) {...}
public void foo(Object... baz) {...}
public void foo(int... baz) {...}
```

Arguments are vastly simpler than the return types because they do not need to get futurized. 

However, there are two potential problem area with these argument types. 

1. How the message ID’s (for the capsule queues) get serialized - is it enough?. This just needs to get tested.
2. Variable-type argument can be primitive (4th example above). However, there comes a time when we forward (or delegate) these arguments from a concrete capsule to its template, and you cannot pass an array of primitives as a substitute for a variable-length argument:

```java
public void foo(int... baz) {...}
foo([1, 2, 3, 4]); // Does not work
```

**Comment by dwtj** [comment]
If we choose to make all variable lengths arguments to procedures illegal, I think this should be rolled into issue #52.

**Comment by dwtj** [comment]
For now, we've decided to make variable length arguments illegal and to add a static check to enforce this. Providing support for them (i.e. making them legal) is now an enhancement: Issue #57.

**Comment by dwtj** [comment]
Duplicate of #60, which was closed by b68b47b8fb11ca594e48fc86f23713e6bf030c68.
